app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/partnership/PartnershipOtherPartnerContactNameController.scala

```
@@ -54,80 +54,80 @@ class PartnershipOtherPartnerContactNameController @Inject() (
-                     partnershipRoutes.PartnershipOtherPartnerContactNameController.submit()
+      inflightOtherPartner.flatMap(_.organisationName).map { organisationName =>
+        MemberName.form()
+          .bindFromRequest()
+          .fold(
+            (formWithErrors: Form[MemberName]) =>
+              Future.successful(
+                BadRequest(
+                  page(formWithErrors,
+                       organisationName,
+                       partnershipRoutes.PartnershipPartnersListController.displayPage(),
+                       partnershipRoutes.PartnershipOtherPartnerContactNameController.submit()
+                  )
                 )
-              )
-            ),
-          fullName =>
-            updateRegistration(fullName).map {
-              case Right(_) =>
-                FormAction.bindFromRequest match {
-                  case SaveAndContinue =>
-                    Redirect(routes.PartnershipOtherPartnerEmailAddressController.displayPage())
-                  case _ =>
-                    Redirect(
-                      partnershipRoutes.PartnershipOtherPartnerContactNameController.displayPage()
-                    )
-                }
-              case Left(error) => throw error
-            }
-        )).getOrElse(Future.successful(NotFound))
+              ),
+            fullName =>
+              updateRegistration(fullName).map {
+                case Right(_) =>
+                  FormAction.bindFromRequest match {
+                    case SaveAndContinue =>
+                      Redirect(routes.PartnershipOtherPartnerEmailAddressController.displayPage())
+                    case _ =>
+                      Redirect(
+                        partnershipRoutes.PartnershipOtherPartnerContactNameController.displayPage()
+                      )
+                  }
+                case Left(error) => throw error
+              }
+          )
+      }.getOrElse(Future.successful(NotFound))
```

I think we should throw IllegalStateExceptions rather than returning 404s. I think that's more consistent with what's going on elsewhere. Not a showstopper but it feels a little "different" to the other controllers.